src/jdk.jfr/share/classes/jdk/jfr/internal/instrument/FileChannelImplInstrumentor.java

```diff
@@ -158,19 +192,26 @@ public int write(ByteBuffer src) throws IOException {
     @JIInstrumentationMethod
     public int write(ByteBuffer src, long position) throws IOException {
         FileWriteEvent event = FileWriteEvent.EVENT.get();
+        FileWriteIOStatisticsEvent writePeriodicEvent = FileWriteIOStatisticsEvent.EVENT.get();
+        int bytesWritten = 0;
+
+        long startTime = System.nanoTime();
         if (!event.isEnabled()) {
             return write(src, position);
```

Comment:
same here

Reply:
Thanks for pointing it out. I've now updated the code to capture read bytes/write bytes even when the 'fileReadEvent'/'fileWriteEvent' is disabled.